terraform/modules/network/vcn-config/nsg_security_rule.tf

```diff
@@ -61,14 +61,32 @@ resource "oci_core_network_security_group_security_rule" "wls_ingress_security_r
 }
 
 resource "oci_core_network_security_group_security_rule" "wls_ingress_internal_security_rule" {
+  count                     = var.configure_secure_mode ? 0 : 1
+  network_security_group_id = element(var.nsg_ids["managed_nsg_id"], 0)
+  direction                 = "INGRESS"
+  protocol                  = "6"
 
+  source      = var.wls_subnet_cidr
+  source_type = "CIDR_BLOCK"
+  stateless   = false
+}
+
+resource "oci_core_network_security_group_security_rule" "wls_ingress_spm_internal_security_rule" {
```

Comment:
I prefer wls_ingress_internal_security_rule_secure_mode to wls_ingress_spm_internal_security_rule

Reply:
I have made the changes